server/events/vcs/gitlab_client.go

```diff
@@ -49,6 +49,11 @@ func NewGitlabClient(hostname string, token string, logger *logging.SimpleLogger
 
 	// If not using gitlab.com we need to set the URL to the API.
 	if hostname != "gitlab.com" {
+		//URL.Parse only works on URLs (i.e.: we need a scheme)
```

Comment:
Let's actually create a new variable here `absoluteURL` rather than rewriting `hostname` which ends up not being a hostname anymore.
```go
absoluteURL := hostname
if !strings.HasPrefix(hostname, "http://") && !strings.HasPrefix(hostname, "https://") {
  absoluteURL = "https://" + hostname
}
```

I don't think we need the comment about url.Parse since we also need the scheme for the base URL later. What I would add a comment about is that we're assume they're using `https` if they haven't specified a scheme.